packages/kit/src/core/postbuild/analyse.js

```diff
@@ -75,39 +96,30 @@ for (const route of manifest._.routes) {
 		}
 
 		if (page) {
+			methods.add('GET');
+			if (page.server?.actions) methods.add('POST');
+
 			validate_page_server_exports(page.server, route.id);
 			validate_common_exports(page.universal, route.id);
 		}
 
 		const should_prerender = get_option(nodes, 'prerender');
-		const prerender =
+		prerender =
 			should_prerender === true ||
 			// Try prerendering if ssr is false and no server needed. Set it to 'auto' so that
 			// the route is not removed from the manifest, there could be a server load function.
 			// People can opt out of this behavior by explicitly setting prerender to false
 			(should_prerender !== false && get_option(nodes, 'ssr') === false && !page?.server?.actions
 				? 'auto'
 				: should_prerender ?? false);
-
-		prerender_map.set(route.id, prerender);
 	}
+
+	metadata.routes.set(route.id, {
```

Comment:
This could be wrong since `prerender` for `route.endpoint` could be a different value than for `route.page`. It's probably almost never going to occur in practice because you'd need a `GET` endpoint next to a page with different `prerender` settings, but it _could_ happen.

Reply:
Yeah, I spotted the same thing and glossed over it. It's not a new bug though, I'm pretty sure the same thing happens now. In any case it's an application error — having a prerendered page and a `GET` in the same place is a bug, so we should probably just throw in that case. Probably best to have a separate issue/PR for that